### PR TITLE
shell bug pull request

### PR DESCRIPTION
**Pull request by tomerlahiyani**

### CODE REVIEW

File: regipy/plugins/ntuser/shellbags_ntuser.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            item_type = "Volume"
+
+        elif isinstance(shell_item, pyfwsi.file_entry):
+            item_type = "File Entry"
+
+        elif isinstance(shell_item, pyfwsi.network_location):
+            item_type = "Network Location"
+
+        else:
+            item_type = 'UNKOWN'
```
Comment (mkorman90) [inline regipy/plugins/ntuser/shellbags_ntuser.py:44]:
```suggestion
            item_type = 'unknown'
```

File: regipy/plugins/ntuser/shellbags_ntuser.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            if long_name:
+                path_segment = long_name
+            elif shell_item.name:
+                path_segment = shell_item.name
+
+        elif isinstance(shell_item, pyfwsi.network_location):
+            if shell_item.location:
+                path_segment = shell_item.location
+
+        if path_segment is None and shell_item.class_type == 0x00:
```
Comment (mkorman90) [inline regipy/plugins/ntuser/shellbags_ntuser.py:78]: Why are you checking this condition if you are not doing anything with it?

File: regipy/plugins/ntuser/shellbags_ntuser.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                    if len(item.extension_blocks) > 0:
+                        for extension_block in item.extension_blocks:
+                            if isinstance(extension_block, pyfwsi.file_entry_extension):
+                                creation_time = extension_block.get_creation_time().isoformat()
+                                access_time = extension_block.get_access_time().isoformat()
+
+                    if hasattr(item,'modification_time'):
+                        modification_time = item.get_modification_time().isoformat()
+                    else:
+                        modification_time = ''
```
Comment (mkorman90) [inline regipy/plugins/ntuser/shellbags_ntuser.py:121]:
```suggestion
                        modification_time = None
```

File: regipy/plugins/ntuser/shellbags_ntuser.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            self.iter_sk(shellbag_ntuser_subkey, NTUSER_SHELLBAG)
+
+
+
+        except RegistryKeyNotFoundException as ex:
+            logger.error(f'Could not find {self.NAME} plugin data at: {NTUSER_SHELLBAG}: {ex}')
+
+
+
+        """
```
Comment (mkorman90) [inline regipy/plugins/ntuser/shellbags_ntuser.py:158]: Delete commented lines

File: regipy_tests/plugin_tests.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import pytest
-
+import datetime
```
Comment (mkorman90) [inline regipy_tests/plugin_tests.py:2]: datetime is usually (as a convention) imported as:
```
import datetime as dt
```

and then used like this: 
```
 dt.datetime(2021, 8, 16, 9, 43, 22).isoformat()
```

**Review by mkorman90 — CHANGES_REQUESTED** [review]
Looks great! Only requiring some minor changes. 

Also, do not forget to add `libfwsi-python==20210419 ` as a dependency in `requirements.txt`

**Review by mkorman90 — APPROVED** [review]

**Review by mkorman90 — APPROVED** [review]

**Review by mkorman90 — APPROVED** [review]

**Comment by SMillerDev** [comment]
It seems this change missed `setup.py` for the new dependency. So anything using that for dependencies (e.g Homebrew) now needs to add it manually.

**Pull request merged**